frontend/src/components/Board/Board.js

```diff
@@ -21,26 +42,57 @@ export default class Board extends React.Component {
     }));
   };
 
+  updateBoard = (row, column, value) => {
+    this.setState(prev => ({
+      boardModel: _.set(
+        prev.boardModel,
+        `rows['${row}'].['${column}'].value`,
+        value
+      )
+    }));
+    this.hideSuggestions();
+  };
+
   render() {
+    // const pressedFieldStyle = {backround: this.state.suggestions ? "green" : "purple"};
     const rows = this.state.boardModel.rows.map((row, idx) => {
       return (
         <tr key={idx}>
           {row.map(field => (
-            <td key={field.col}>
-              <Field
-                row={field.row}
-                col={field.col}
-                value={field.value}
-                onDrop={item => this.handleDrop(field.row, field.col, item)}
-              />
-            </td>
+            <LongPress
+              key={field.col}
+              time={200}
+              onLongPress={() => this.displaySuggestions(field.row, field.col)}
+              onPress={() => this.hideSuggestions()}
+            >
+              <td key={field.col} id={`${field.row}x${field.col}`}>
+                <Field
+                  // style = {pressedFieldStyle}
+                  row={field.row}
+                  col={field.col}
+                  value={field.value}
+                  onDrop={item => this.handleDrop(field.row, field.col, item)}
+                />
+              </td>
+            </LongPress>
           ))}
         </tr>
       );
     });
 
     return (
       <div className="sudoku sudoku-background">
+        {this.state.suggestions ? (
```

Comment:
Instead of ternary operators, use short circuits - they look a little cleaner. `{this.state.suggestions && (<CircularMenu ... />)}`